ethers-providers/src/rpc/transports/mock.rs

```diff
@@ -72,6 +83,31 @@ impl JsonRpcClient for MockProvider {
     }
 }
 
+impl PubsubClient for MockProvider {
```

Comment:
I believe these implementations will need to be gated with `not(target_arch = "wasm32")`, see CI